mne/io/cnt/cnt.py

```diff
@@ -0,0 +1,404 @@
+            fid.seek(event_offset + 9 + i * event_bytes + 4)
+            offset = np.fromfile(fid, dtype='<i4', count=1)[0]
+            event_time = (offset - 900 - 75 * n_channels) // (n_channels * 2)
+            stim_channel[event_time - 1] = event_id
+
+    info = _empty_info(sfreq)
+    if lowpass_toggle == 1:
+        info['lowpass'] = highcutoff
+    if highpass_toggle == 1:
+        info['highpass'] = lowcutoff
+    subject_info = {'hand': hand, 'id': patient_id, 'sex': sex,
+                    'first_name': first_name, 'last_name': last_name}
+
+    if eog == 'auto':
+        eog = [ch for ch in ch_names if 'EOG' in ch.upper()]
+    if ecg == 'auto':
+        ecg = [ch for ch in ch_names if 'ECG' in ch.upper()]
+    if emg == 'auto':
+        emg = [ch for ch in ch_names if 'EMG' in ch.upper()]
+    eegs = list()
+    for idx, ch_name in enumerate(ch_names):
+        if ch_name in eog or idx in eog:
+            coil_type = FIFF.FIFFV_COIL_NONE
+            kind = FIFF.FIFFV_EOG_CH
+        elif ch_name in ecg or idx in ecg:
+            coil_type = FIFF.FIFFV_COIL_NONE
+            kind = FIFF.FIFFV_ECG_CH
+        elif ch_name in emg or idx in emg:
+            coil_type = FIFF.FIFFV_COIL_NONE
+            kind = FIFF.FIFFV_EMG_CH
+        elif ch_name in misc or idx in misc:
+            coil_type = FIFF.FIFFV_COIL_NONE
+            kind = FIFF.FIFFV_MISC_CH
+        else:
+            coil_type = FIFF.FIFFV_COIL_EEG
+            kind = FIFF.FIFFV_EEG_CH
+            eegs.append(idx)
+
+        chan_info = {'cal': cals[idx], 'logno': idx + 1, 'scanno': idx + 1,
+                     'range': 1.0, 'unit_mul': 0., 'ch_name': ch_name,
+                     'unit': FIFF.FIFF_UNIT_V,
+                     'coord_frame': FIFF.FIFFV_COORD_HEAD,
+                     'coil_type': coil_type, 'kind': kind}
+        chs.append(chan_info)
+
+    coords = _topo_to_sphere(pos, eegs)
+    locs = np.zeros((len(chs), 12), dtype=float)
+    locs[:, :3] = coords
+    for ch, loc in zip(chs, locs):
+        ch.update(loc=loc)
+
+    # Add the stim channel.
+    chan_info = {'cal': 1.0, 'logno': len(chs) + 1, 'scanno': len(chs) + 1,
+                 'range': 1.0, 'unit_mul': 0., 'ch_name': 'STI 014',
+                 'unit': FIFF.FIFF_UNIT_NONE,
+                 'coord_frame': FIFF.FIFFV_COORD_UNKNOWN, 'loc': np.zeros(12),
+                 'coil_type': FIFF.FIFFV_COIL_NONE, 'kind': FIFF.FIFFV_STIM_CH}
+    chs.append(chan_info)
+    baselines.append(0)  # For stim channel
+    cnt_info.update(baselines=np.array(baselines), n_samples=n_samples,
+                    stim_channel=stim_channel)
+    info.update(filename=input_fname, meas_date=np.array([meas_date, 0]),
+                description=str(session_label), buffer_size_sec=10., bads=bads,
+                subject_info=subject_info, chs=chs)
+    info._check_consistency()
+    return info, cnt_info
+
+
+def _topo_to_sphere(pos, eegs):
```

Comment:
this function is basically undoing the 2d proj we have to do to plot topos. It basically reverts it. Can you put this function next to the one that does this 3d to 2d transform?